stratum/hal/bin/ipdk/ipdk_main.cc

```diff
@@ -0,0 +1,119 @@
+// Copyright 2018-2019 Barefoot Networks, Inc.
+// Copyright 2020-present Open Networking Foundation
+// Copyright 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "stratum/hal/bin/ipdk/ipdk_main.h"
+
+#include "gflags/gflags.h"
+#include "stratum/glue/init_google.h"
+#include "stratum/glue/logging.h"
+#include "stratum/hal/lib/ipdk/ipdk_chassis_manager.h"
+#include "stratum/hal/lib/ipdk/ipdk_hal.h"
+#include "stratum/hal/lib/ipdk/ipdk_node.h"
+#include "stratum/hal/lib/ipdk/ipdk_switch.h"
+#include "stratum/hal/lib/phal/phal_sim.h"
+#include "stratum/hal/lib/tdi/tdi_action_profile_manager.h"
+#include "stratum/hal/lib/tdi/tdi_counter_manager.h"
+#include "stratum/hal/lib/tdi/tdi_pre_manager.h"
+#include "stratum/hal/lib/tdi/tdi_sde_wrapper.h"
+#include "stratum/hal/lib/tdi/tdi_table_manager.h"
+#include "stratum/lib/security/auth_policy_checker.h"
+
+DEFINE_string(ipdk_sde_install, "/usr",
+              "Absolute path to the directory where the SDE is installed");
+DEFINE_bool(ipdk_infrap4d_background, false,
+            "Run infrap4d in the background with no interactive features");
+// TODO(dfoster): Default value for IPDK?
+DEFINE_string(ipdk_infrap4d_cfg, "stratum/hal/bin/ipdk/tofino_skip_p4.conf",
+              "Path to the infrap4d json config file");
+
+namespace stratum {
+namespace hal {
+namespace barefoot {
```

Comment:
Please change namespace to ipdk everywhere

Reply:
IPDK will probably be part of the Stratum `tdi` namespace once the latter has been defined.